pkg/provider/provider.go

```diff
@@ -179,6 +181,20 @@ func (p *Provider) buildServicesAndRoutersForService(t *topology.Topology, cfg *
 			cfg.TCP.Services[key] = p.buildTCPServiceFromService(t, svc, svcPort.TargetPort.IntVal)
 			cfg.TCP.Routers[key] = buildTCPRouter(rule, entrypoint, key)
 		}
+
+	case k8s.ServiceTypeUDP:
+		for _, svcPort := range svc.Ports {
+			entrypoint, err := p.buildUDPEntrypoint(svc, svcPort.Port)
+			if err != nil {
+				p.logger.Errorf("Unable to build UDP entrypoint for Service %q and port %d: %v", svcKey, svcPort.Port, err)
```

Comment:
If I understand this correctly. When creating the shadow service, UDP ports have been saved into the UDP state table. And so, while iterating on ports here, if the port is not UDP, we are just not gonna find it in the UDP state table. That's how ports get filtered. Could you confirm?

Reply:
Yes.